libraries/Microsoft.Bot.Builder.Core/Middleware/StateMiddleware.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Middleware;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Middleware
+{
+    public class StateSettings
+    {
+        public bool WriteBeforeSend { get; set; } = true;
+        public bool LastWriterWins { get; set; } = true;
+    }
+
+    public abstract class StateMiddleware<StateT> : IContextCreated, ISendActivity
```

Comment:
>StateMiddleware [](start = 26, length = 15)

I'm working on a similar base class called BotState.  adding Middleware on the end here seems a bit redundant given that you're in the Middleware namespace.

Reply:
I changed this to Botstate

---
In reply to: [170473219](https://github.com/Microsoft/botbuilder-dotnet/pull/193#discussion_r170473219) [](ancestors = 170473219)